cpp/src/groupby/hash/compute_shared_memory_aggs.hpp

```diff
@@ -22,15 +22,15 @@
 #include <rmm/cuda_stream_view.hpp>
 
 namespace cudf::groupby::detail::hash {
-std::size_t get_available_shared_memory_size(cudf::size_type grid_size);
+size_type get_available_shared_memory_size(cudf::size_type grid_size);
 
-std::size_t constexpr compute_shmem_offsets_size(cudf::size_type num_cols)
+size_type constexpr compute_shmem_offsets_size(cudf::size_type num_cols)
```

Comment:
What are all these size changes for? I would have expected the previous code to be safe. Was there a problem with downcasting for pointer arithmetic, or using an unsigned type?

Reply:
In the current code, where both the available size and the offset size are of type `size_t`, an issue arises when the offset size exceeds the available size. Since `size_t` is an unsigned type, this results in a very large number instead of a negative value. Changing these types to signed integers can resolve this issue.

Reply:
Got it.